libgraph/src/analytics/k_shortest_paths/ksssp.cpp

```diff
@@ -0,0 +1,464 @@
+
+    if (reachable) {
+      switch (plan.algorithm()) {
+      case kSsspPlan::kDeltaTile:
+        DeltaStepAlgo<kSSSPSrcEdgeTile, OBIM>(
+            &edge_data, &graph, source, kSSSPSrcEdgeTilePushWrap{&graph},
+            kSSSPTileRangeFn(), &paths, &path_pointers, path_alloc, report_node,
+            num_paths, step_shift);
+        break;
+      case kSsspPlan::kDeltaStep:
+        DeltaStepAlgo<kSSSPUpdateRequest, OBIM>(
+            &edge_data, &graph, source, kSSSPReqPushWrap(),
+            kSSSPOutEdgeRangeFn{&graph}, &paths, &path_pointers, path_alloc,
+            report_node, num_paths, step_shift);
+        break;
+      case kSsspPlan::kDeltaStepBarrier:
+        katana::gInfo("Using OBIM with barrier\n");
+        DeltaStepAlgo<kSSSPUpdateRequest, OBIM_Barrier>(
+            &edge_data, &graph, source, kSSSPReqPushWrap(),
+            kSSSPOutEdgeRangeFn{&graph}, &paths, &path_pointers, path_alloc,
+            report_node, num_paths, step_shift);
+        break;
+
+      default:
+        return katana::ErrorCode::InvalidArgument;
+      }
+    }
+
+    execTime.stop();
+    page_alloc.Report();
+
+    if (reachable) {
+      std::multimap<uint32_t, Path*> paths_map;
+
+      for (auto pair : paths) {
+        paths_map.insert(std::make_pair(pair.first, pair.second));
+      }
+
+      katana::gPrint("Node ", report, " has these k paths:\n");
+
+      uint32_t num =
+          (paths_map.size() > num_paths) ? num_paths : paths_map.size();
+
+      auto it_report = paths_map.begin();
+
+      for (uint32_t iter = 0; iter < num; iter++) {
+        const Path* path = it_report->second;
+        PrintPath(path);
+        katana::gPrint(" ", report, "\n");
+        katana::gPrint("Weight: ", it_report->first, "\n");
+        it_report++;
+      }
+
+      katana::do_all(katana::iterate(path_pointers), [&](Path* p) {
+        path_alloc.DeletePath(p);
+      });
+    }
+
+    return katana::ResultSuccess();
+  }
+};
+
+template <typename GraphTy, typename Weight>
+katana::Result<void>
+Ksp(GraphTy& pg, unsigned int start_node, unsigned int report_node,
+    AlgoReachability algo_reachability, unsigned int num_paths,
+    unsigned int step_shift, kSsspPlan plan) {
+  static_assert(std::is_integral_v<Weight> || std::is_floating_point_v<Weight>);
+  SsspImplementation<GraphTy, Weight> impl{{plan.edge_tile_size()}};
+  return impl.KSP(
+      pg, start_node, report_node, algo_reachability, num_paths, step_shift,
+      plan);
+}
+
+template <typename Weight>
+static katana::Result<void>
+kSSSPWithWrap(
+    katana::PropertyGraph* pg, unsigned int start_node,
+    unsigned int report_node, katana::TxnContext* txn_ctx,
+    AlgoReachability algo_reachability, unsigned int num_paths,
+    unsigned int step_shift, kSsspPlan plan) {
+  auto result =
+      ConstructNodeProperties<std::tuple<NodeCount, NodeMax>>(pg, txn_ctx);
+  if (!result) {
+    return result.error();
+  }
+
+  using GraphTy = katana::TypedPropertyGraph<
```

Comment:
oh you are still using the property graph format; you need to change it to views format. look at how k-core app is using views.